hikari/internal/enums.py

```diff
@@ -58,6 +89,11 @@ def __setitem__(self, name: str, value: typing.Any) -> None:
         if name == "" or name == "mro":
             raise TypeError(f"Invalid enum member name: {name!r}")
 
+        if isinstance(value, deprecated):
+            value = _DeprecatedAlias(name, self.values_to_names[value.value], value.removal_version)
+            super().__setitem__(name, value)
```

Comment:
This behaviour seems to need testing (both for the type checking and runtime behaviour tbh)